// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";
import "google/protobuf/empty.proto";
// option   go_package = "/routeguide";
// option java_multiple_files = true;
// option java_package = "io.grpc.examples.routeguide";
// option java_outer_classname = "RouteGuideProto";

package service.grpc;

// Interface exported by the server.
service CameraService {

  // get all service meta info
  rpc GetCameraMetas(google.protobuf.Empty) returns (CameraMetas) {}

  // input camera sn , return 1 frame , alive image
  rpc GetImage(ImageRequest_) returns (BytesBody) {}


  rpc Reset( google.protobuf.Empty ) returns (google.protobuf.Empty) {}

}

message CameraMeta_ {
  string serialNumber = 1;
  string modelName = 2;
  string manufactureName = 3;
  string deviceVersion = 4;
  string userDefinedName = 5;
  string cameraType = 6;
  map<string, string> info = 7;
  // for ext...
}

message CameraMetas { repeated CameraMeta_ data = 1; }

message BytesBody { bytes content = 1; }

message ImageRequest_ {
  string sn = 1;
  ImageShape_ imageShape = 2;
}

message ImageShape_ {
  int32 width = 1;
  int32 height = 2;
  int32 channel = 3;
}